packages/tasks/src/tasks/chat-completion/spec/input.json

```diff
@@ -47,17 +51,42 @@
 			"type": "number",
 			"description": "The value used to modulate the logits distribution."
 		},
+		"tools": {
+			"type": "array",
+			"items": {
+				"$ref": "#/definitions/Tool"
+			},
+			"description": "The tools to use for generation."
+		},
+		"tool_choice": {
+			"type": "string",
+			"description": "The tool to use for generation."
+		},
 		"top_p": {
 			"type": "number",
 			"description": "If set to < 1, only the smallest set of most probable tokens with probabilities that add up to `top_p` or higher are kept for generation."
 		}
 	},
-	"required": ["messages"],
 	"definitions": {
 		"Role": {
 			"oneOf": [{ "const": "assistant" }, { "const": "system" }, { "const": "user" }],
 			"title": "ChatCompletionMessageRole",
 			"description": "The role of the message author."
+		},
+		"Tool": {
+			"title": "ChatCompletionTool",
+			"description": "The tool to use for generation.",
+			"properties": {
+				"type": {
+					"type": "string",
+					"description": "The type of tool to use."
+				},
+				"function": {
+					"type": "object",
```

Comment:
Does `function` always look like `name: str, description: Optional[str], parameters: object`? (taken from [here](https://platform.openai.com/docs/api-reference/chat/create)) or is it different? If always the same, let's add it to the specs.